### PR TITLE
Game Timer: domain tests and UI/store thinning (#218)

### LINKED ISSUE #218: Game Timer: domain tests and UI/store thinning
## Problem Statement

Game Timer’s authoritative **snapshot** rules (turn selection, **banked time**, **hard pass**, **round** navigation, per-**round** **player** order) live largely in the Pinia store alongside persistence and sync orchestration, with little dedicated coverage at the pure-function layer. Display derivations (progress ratios, **timing strip** totals, multi-**round** UI mode) are split between store getters and a large **player** list component. Movie Vote received a comparable deepening pass (#217); Game Timer did not.

Facilitators depend on correct timer behavior at the table, but this slice is a **strict refactor**: no facilitator-visible behavior or UX changes unless something is seriously broken (data-loss, sync corruption, or contradiction with documented domain language).

**Parent:** #211

## Solution

Extract authoritative timer **snapshot** mutations into tested pure functions, move display derivations into tested pure helpers, introduce a role-agnostic **player** list view model (numbers and flags only), and thin the store to persistence + sync + thin orchestration—mirroring the Movie Vote results deepening pattern without changing what collaborators see or how the table plays.

## User Stories

1. As a facilitator hosting a **room**, I want turn timing and **banked time** to behave exactly as before, so that I can trust the timer during a long tabletop session.
2. As a **guest** mirroring the **host**, I want **snapshot** fields to remain the sync contract, so that my device stays aligned without treating local display math as truth.
3. As a facilitator, I want tap-to-select / tap-again-to-pause on the **active player** to work unchanged, so that I can pause mid-thought without losing accrued time incorrectly.
4. As a facilitator, I want **end turn** to advance in list order and skip **hard pass**ed **players** in the current **round**, so that passed players stay out of the rotation until the **round** changes.
5. As a facilitator with **hard pass** enabled, I want passing a **player** to bank any running clock and auto-advance when it was their **turn**, so that the table keeps moving.
6. As a facilitator, I want undoing a **hard pass** to remove pass eligibility without restoring time already banked during a running clock, so that mistakes are fixable without rewriting history.
7. As a facilitator, I want everyone **hard pass**ing in a **round** to auto-advance to the next **round**, so that an exhausted **round** resolves without manual navigation.
8. As a facilitator with **pass order determines round order** enabled, I want passers to lead the next **round**’s order, so that seating dynamics carry forward as today.
9. As a facilitator navigating **rounds**, I want live **turns** paused and per-**round** order applied without wiping lifetime **banked time**, so that earlier **rounds** still count toward totals.
10. As a facilitator, I want **new game with same players** to reset clocks and **round** state while keeping roster and session toggles, so that we can start fresh without re-entering names.
11. As a facilitator, I want removing the **active player** to bank their live segment and clear the live **turn**, so that totals stay honest.
12. As a facilitator viewing the **player** list, I want per-**round** and lifetime progress bars and time readouts to match today’s values, so that the cleanup pass does not change what I see.
13. As a facilitator using the **timing strip**, I want total and non-**player** elapsed modes to match today’s math, so that session pacing readouts stay familiar.
14. As a **guest**, I want the same row times and styling as the **host** (with host-only gestures still gated), so that read-only mirroring stays consistent.
15. As a maintainer, I want authoritative **snapshot** transitions unit-tested as pure functions, so that refactors cannot silently break tabletop rules.
16. As a maintainer, I want display derivations tested separately from sync, so that bar widths and strip totals are not confused with the multiplayer contract.
17. As a maintainer, I want the **player** list view model to expose numbers and flags only, so that tests stay stable when copy or theme styling changes.
18. As a maintainer, I want existing Game Timer unit tests to keep passing, so that P2P facade coverage (#214) and session tests remain valid.
19. As a maintainer, I want ESLint clean on touched modules, so that the architecture epic bar (#211) stays met.
20. As a facilitator, I do not want **keep display on**, star-room core, or Nosleep behavior touched, so that unrelated ADR decisions stay stable.

### Behavior preservation bar

- **Strict refactor only.** No facilitator-visible behavior or UX changes unless a finding is (1) objectively incorrect against `CONTEXT.md` plus observable table behavior, or (2) data-loss / **snapshot** sync-corruption class. Otherwise extract and test current behavior as-is; file follow-ups for desired rule changes.

### Authoritative rules vs display derivation

- **Snapshot mutations** (fields in `GameTimerSyncPayload`: **players**, **active player**, live **turn** timestamps, **round**, order maps, **hard pass** flags and lists) are authoritative and must be implemented as pure, unit-tested functions the store calls.
- **Display derivations** (progress ratios, formatted durations, **timing strip** totals, multi-**round** UI mode, row active/paused/**hard pass**ed flags) are computed from **snapshot** + wall clock locally—not separately synced. `CONTEXT.md` **Snapshot** entry documents this boundary.

### Pure timer rule module

- Extract store action logic for must-lock flows into pure functions (immutable-in / new-state-out or equivalent explicit return shape).
- Must-lock flows:
  1. **Turn** selection — start clock; tap same **active player** pauses (banks segment, holds selection); tap again resumes.
  2. **Turn** advance — wrap list order; skip **hard pass**ed **players** in current **round**.
  3. **Hard pass** — bank running clock; record pass order; auto-advance if it was their **turn**; undo removes pass without restoring banked time.
  4. All **hard pass**ed in **round** → next **round**.
  5. **Pass order determines round order** → draft next-**round** order from pass sequence.
  6. **Round** navigation — pause live **turn**; apply saved per-**round** order; preserve lifetime **banked time**.
  7. **New game with same players** — roster + toggles survive; clocks/**round**/**hard pass** state reset.
  8. Remove **active player** — bank segment; clear live **turn**; strip from order maps.

### Pure display helpers

- Move `hasMultipleRounds`, `totalGameElapsedMs`, and `nonPlayerElapsedMs` from store getters into pure helpers alongside existing `displayedMsForPlayer`, `progressRatio`, etc.
- Store retains orchestration: call helpers, persist, delegate mutations to rule functions.

### Player list view model

- Extract the `playerRowsById` computation into a `createPlayerListViewModel` (or equivalent) factory.
- **Role-agnostic:** inputs are **snapshot** slice + `nowMs` + multi-**round** flag; outputs per-row numbers (`displayedMs`, `displayedMsRound`, `progress`, `progressRound`) and flags (`isActive`, `isHardPassed`, `isPausedHeldTurn`).
- A thin composable wires reactive store state + `useGameTimerNow` tick into the factory (Movie Vote `useMovieVoteResultsPresentation` pattern).
- Shell component keeps: drag-reorder, swipe edit/delete dialogs, scroll-active-row-into-view, host-only mutation gating via `isGuest`, `formatDurationMs` labels, and `playerBar*` color helpers.

### Store thinning

- Pinia store becomes: state shape, persistence/hydration normalization, calling pure rule functions, and sync-facing action names consumed by the P2P bridge.
- Public action names and `GameTimerSyncPayload` wire shape unchanged unless a serious bug forces a fix (separate escalation).

### Scope of UI extraction

- **Player list only** for view-model extraction this slice.
- **Turn** controls and **round** bar remain as-is (thin enough; no new view models required for acceptance).

### P2P and persistence

- Do not expand P2P facade contract tests (#214 owns that seam).
- Persistence `afterHydrate` normalization may remain in the store; lower priority for new pure tests unless touched during extraction.

### What makes a good test here

- Assert **external behavior**: resulting **snapshot** fields, banked millisecond totals, **round** index, order arrays, and view-model numeric outputs.
- Do **not** assert user-facing copy, formatted time strings, CSS classes, or theme colors.
- Use fake `Date.now` (or injected `nowMs`) for time-dependent rules and derivations.

### Testing seams (highest first)

1. **Pure timer rule functions** — authoritative **snapshot** transitions for must-lock flows 1–8.
2. **Pure display helpers** — `displayedMs*`, `progressRatio`, `hasMultipleRounds`, timing-strip elapsed math.
3. **Player list view model factory** — row map: ms totals, ratios, active/paused/**hard pass** flags; stable when only formatting would differ.
4. **Pinia store integration** — migrate or retain minimal tests proving store still orchestrates correctly (prior `startNewGameSamePlayers` and session timing tests).
5. **Vue shells** — not in scope; no component snapshot tests.

### Prior art

- Movie Vote #217: `create*ViewModel` factories + presentation composable; ballot helpers moved into `core.js`; store thinned.
- Existing Game Timer store tests with `withFakeNow` pattern.
- Game Timer P2P facade contract tests (#214) — remain green; not expanded here.
- Project rule: unit tests avoid locking presentation strings.

## Out of Scope

- Nosleep / **keep display on** behavior (ADR 0002).
- Star-room session core changes.
- Full Vue component test suite for every Game Timer shell.
- Turn controls and **round** bar view-model extraction (unless discovered necessary for acceptance).
- P2P wire surface expansion (#214).
- Facilitator-visible UX polish, copy changes, or new features.
- Lower-priority rule edge cases unless extraction naturally covers them: add-**player** mid-**round** order append, legacy hydrate migration paths, empty-roster no-ops.

## Further Notes

- Complements #214 (facade contract tests); this issue owns **core** timer rules and **UI/store** boundaries.
- Epic: #211 architecture clean pass (slice 7).
- `CONTEXT.md` **Snapshot** clarifies that progress bars, formatted durations, and **timing strip** totals are local derivations—not synced fields.
- Acceptance checklist from original issue remains: dedicated `core`/rule tests, at least one major component display logic in tested view model, reduced store surface without behavior change, existing tests green, ESLint clean on touched files.

### PR DESCRIPTION
**Pull request by enmaku**
## Summary

- Extract authoritative Game Timer snapshot mutations into unit-tested pure functions in `timerRules.js`, and move display derivations (`hasMultipleRounds`, timing-strip elapsed math) into `core.js`.
- Introduce `createPlayerListViewModel` plus `useGameTimerPlayerListPresentation` so the player list shell keeps drag/swipe/formatting while row ms, ratios, and flags come from a tested view model.
- Thin the Pinia store to persistence, sync orchestration, and delegation; fix host swipe-to-edit/delete by limiting drag-reorder to the turn column handle.

Closes #218. Part of #211.

## Test plan

- [x] `npm test -- src/features/game-timer/timerRules.test.js src/features/game-timer/core.test.js src/features/game-timer/ui/createPlayerListViewModel.test.js`
- [x] `npm test -- src/features/game-timer/startNewGameSamePlayers.test.js src/features/game-timer/sessionTiming.test.js`
- [x] `npm test -- src/features/game-timer/p2p/*.test.js`
- [x] ESLint on touched files
- [x] Manual: host swipe right on player name → edit dialog; swipe left → delete confirm; long-press left turn column → reorder
- [x] Manual: turn select/pause, hard pass, round nav, timing strip totals unchanged from pre-refactor behavior

### CODE REVIEW

File: src/features/game-timer/composables/useGameTimerPlayerListPresentation.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      },
+      round: round.value,
+      hardPassEnabled: hardPassEnabled.value,
+      hardPassOrderByRound: store.hardPassOrderByRound,
+      hasMultipleRounds: hasMultipleRoundsFlag.value,
+      nowMs: unref(now),
+    }),
+  )
+
+  function isHardPassed(player) {
```
Comment (enmaku) [inline src/features/game-timer/composables/useGameTimerPlayerListPresentation.js]: [structure] `createPlayerListViewModel` already puts `isHardPassed` and `isPausedHeldTurn` on each row (lines 46–60 in the view model). These composable helpers duplicate that logic, and `GameTimerPlayerList` calls them instead of `rowForPlayer(player)?.isHardPassed`. Can we drop the helpers and read the row flags so there is one source of truth for presentation state?

File: src/features/game-timer/ui/createPlayerListViewModel.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const maxMs = maxDisplayedMs(players, session, nowMs)
+  const maxRoundMs = hasMultipleRounds ? maxDisplayedMsInRound(players, session, nowMs, round) : 0
+  const map = new Map()
+
+  for (const p of players) {
+    const displayedMs = displayedMsForPlayer(p, session, nowMs)
+    const displayedMsRound = hasMultipleRounds
+      ? displayedMsForPlayerInRound(p, session, nowMs, round)
+      : 0
+    const isHardPassed = hardPassEnabled && hardPassIdsThisRound.has(p.id)
```
Comment (enmaku) [inline src/features/game-timer/ui/createPlayerListViewModel.js:46]: [architecture] Rows already carry `isHardPassed` / `isPausedHeldTurn` here — good **locality** for derived presentation. The composable re-derives the same predicates separately; wiring the template to these row fields would let this module own the full player-row **interface** and delete ~10 lines upstream.

File: src/features/game-timer/timerRules.js [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    hardPassOrderByRound: state.hardPassOrderByRound,
+    totalGameStartedAt: state.totalGameStartedAt,
+  }
+}
+
+/**
+ * @param {GameTimerRuleSession} session
+ * @returns {Partial<GameTimerRuleSession>}
+ */
+export function ruleSessionPatch(session) {
```
Comment (enmaku) [inline src/features/game-timer/timerRules.js]: [structure] `ruleSessionPatch` is exported but nothing imports it (store uses `applyRuleSessionToStore` with the full session). Delete unless you have a partial-apply caller planned — dead surface on a module you are trying to keep deep.

File: src/stores/gameTimer.js [hunk 4]
```diff
@@ -3,43 +3,21 @@
...
-  for (const k of Object.keys(hardPassOrderByRound)) {
-    const arr = hardPassOrderByRound[k]
-    if (Array.isArray(arr)) {
-      hardPassOrderByRound[k] = arr.filter((id) => id !== playerId)
-    }
-  }
-}
+import { createPlayerId, nextDefaultColor } from '../features/game-timer/core.js'
+import {
+  applyPlayerOrder,
```
Comment (enmaku) [inline src/stores/gameTimer.js]: [architecture] Actions now cross into `timerRules`, but the store still retains `_applyOrderForActiveRound` and `_recomputeNextRoundOrderFromHardPasses` with parallel bodies in `timerRules` (used from `afterHydrate`, `setHardPassOrderNextRound`, and the P2P bridge). That splits rule **locality** across two modules and invites drift. Export the shared helpers from `timerRules` and call them from those store/bridge paths, then delete the store copies (and the now-dead `_saveOrderForRound` / `_nextNonPassedPlayerIndex`).

File: src/stores/gameTimer.js [hunk 5]
```diff
@@ -305,173 +232,43 @@ export const useGameTimerStore = defineStore('gameTimer', {
...
-      }
-
-      const list = this.hardPassOrderByRound[rk] ?? []
-      const passedSet = new Set(list)
-      const allHardPassed =
-        this.players.length > 0 && this.players.every((p) => passedSet.has(p.id))
-      if (allHardPassed) {
-        this.goToNextRound()
-      }
+      const next = registerHardPassSnapshot(ruleSessionFromStoreState(this), playerId, Date.now())
```
Comment (enmaku) [inline src/stores/gameTimer.js:175]: [structure] This delegation pattern is the right end state — thin Pinia **adapter**. Once hydrate/toggle paths also route through `timerRules`, the remaining private `_`-prefixed rule helpers in the store can go entirely.

**Review by enmaku — COMMENTED** [review]
## Verdict
COMMENT — Would approve on substance; posting COMMENT because this is my own PR. The extraction direction is solid and the new tests earn their keep, but a few follow-ups would finish the thinning instead of leaving parallel implementations.

## Thermo-nuclear (diff)
- **Presentation flag duplication:** `createPlayerListViewModel` already computes `isHardPassed` / `isPausedHeldTurn` on each row, but the composable re-implements the same predicates and the list component calls those instead of `rowForPlayer(player)?.isHardPassed`. Same behavior, two sources — easy drift when one path changes.
- **Partial rule extraction:** Turn/round/hard-pass actions now delegate to `timerRules`, but the store still keeps `_applyOrderForActiveRound`, `_recomputeNextRoundOrderFromHardPasses`, `_saveOrderForRound`, and `_nextNonPassedPlayerIndex`. The last two are dead after this diff; the first two duplicate `timerRules` and remain wired from `afterHydrate`, `setHardPassOrderNextRound`, and the P2P bridge — two implementations of the same **snapshot** rules.
- **Dead export:** `ruleSessionPatch` is exported with no callers.

## Architecture (PR-scoped)
- **Deepen the rule seam:** Export `applyOrderForActiveRound` / `recomputeNextRoundOrderFromHardPasses` from `timerRules` (or a single hydrate helper) and delete the store copies so hydrate + toggle paths share the same module as the snapshot actions — restores **locality** at the authoritative-rule **interface**.
- **Collapse presentation locality:** Let `GameTimerPlayerList` read row flags from `playerRowsById` and drop the composable helpers; the view-model module then owns all derived row state behind one **interface**.
- **Adapter stays thin (good):** Store actions as `ruleSessionFromStoreState → *Snapshot → applyRuleSessionToStore` is the right Pinia **adapter** shape; keep pushing until the store holds no rule logic beyond wiring and roster/settings mutations.

## Unanchored notes
- Delete store `_saveOrderForRound` and `_nextNonPassedPlayerIndex` — no remaining call sites after the action delegation.
- `sessionTiming.test.js` / `startNewGameSamePlayers.test.js` still exercise Pinia while `timerRules.test.js` covers the pure layer; fine for now, but consider dropping redundant store-level cases once the adapter is the only path.
